src/rest_vol.c

```diff
@@ -2414,7 +2418,7 @@ RV_convert_dataspace_shape_to_JSON(hid_t space_id, char **shape_body, char **max
                     CHECKED_REALLOC(shape_out_string, shape_out_string_curr_len, (size_t) buf_ptrdiff + MAX_NUM_LENGTH + 1,
                                     shape_out_string_curr_pos, H5E_DATASPACE, FAIL);
 
-                    if ((bytes_printed = sprintf(shape_out_string_curr_pos, "%s%llu", i > 0 ? "," : "", dims[i])) < 0)
+                    if ((bytes_printed = sprintf(shape_out_string_curr_pos, "%s%llu", i > 0 ? "," : "", (long long unsigned int) dims[i])) < 0)
```

Comment:
Rather than converting these to `long long`, we actually have a few printf macro for `hsize_t` values in H5public.h such as `PRIuHSIZE`. So the format here could be `"%s" PRIuHSIZE`.